### DISCUSSION TITLE
Should the graph fit the RAM?

### DISCUSSION THREAD

**Discussion by larytet** [networkx/networkx#5228] [Q&A]
I have 20M nodes where a node can be connected  between 0 to 32K children. Typical number of children is 50. Typical depth is 10, but some trees can higher than 20 .
A node is a Tuple[str, str] where a string is usually shorter than 50 chars

A set containing the nodes doesn't fit the RAM. I use Python Shove package.

Will networkx  handle such graph?

**Comment by dschult** [comment]
How much RAM do you have ;}   

More seriously, it is the number of edges that impacts ram. It looks like 20M nodes with an average degree of 50 makes ~1B edges. Even at 32 bytes per edge that is 32GB.  And 32 bytes is an extremely low estimate of the dict storage that doesn't account for storage required for the node names, edge attributes, etc.  You should definitely convert the node identifier to integers, but even then I'm guessing that to do anything useful you would need a few hundred GB of ram. 

Depending on what you are doing, it may make sense to process part of the network at a time. That can be very effective for saving memory.
